data/pools/transactionPool.go

```diff
@@ -430,6 +437,10 @@ func (pool *TransactionPool) ingest(txgroup []transactions.SignedTxn, params poo
 		return ErrNoPendingBlockEvaluator
 	}
 
+	if pool.shutdown {
```

Comment:
I would think mu has to control access to this right? Noted comment that on lines 433/434 that it may not be locked?

Reply:
`ingest` is always called locked. The comment refers to this line `condvar.TimedWait(&pool.cond, timeoutOnNewBlock)` below where `cond.Wait()` releases the lock.